# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmMemory.h
# Commit: 13b7086f65cb
# Full Hash: 13b7086f65cbbbf1fed6e0341cfbdfb309fb962c
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-09-17 13:36 +0000
# Regressor Bug: 1969509
# File Overlap Count: 6
# Overlapping Files: js/src/vm/ArrayBufferObject.cpp, js/src/wasm/WasmMemory.h, js/src/jit-test/tests/wasm/memory64/basic.js, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer.js, js/src/jit-test/lib/wasm.js
# Description:
#   Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt
# ==============================================================================

diff -r f7c96ac82c1a -r 13b7086f65cb js/src/wasm/WasmMemory.h
--- a/js/src/wasm/WasmMemory.h	Wed Sep 17 13:35:00 2025 +0000
+++ b/js/src/wasm/WasmMemory.h	Wed Sep 17 13:36:10 2025 +0000
@@ -102,6 +102,13 @@
     return length.value();
   }
 
+  // Return the byteLength for a 64-bits memory.
+  uint64_t byteLength64() const {
+    mozilla::CheckedInt<uint64_t> length(value_);
+    length *= PageSize;
+    return length.value();
+  }
+
   // Increment this pages by delta and return whether the resulting value
   // did not overflow. If there is no overflow, then this is set to the
   // resulting value.
